manim/mobject/mobject.py

```diff
@@ -1637,20 +1655,21 @@ def shift_onto_screen(self, **kwargs) -> Self:
                 self.to_edge(vect, **kwargs)
         return self
 
-    def is_off_screen(self):
+    def is_off_screen(self) -> bool:
         if self.get_left()[0] > config["frame_x_radius"]:
             return True
         if self.get_right()[0] < -config["frame_x_radius"]:
             return True
         if self.get_bottom()[1] > config["frame_y_radius"]:
             return True
-        return self.get_top()[1] < -config["frame_y_radius"]
+        rv: bool = self.get_top()[1] < -config["frame_y_radius"]
+        return rv
```

Comment:
MyPy should have been able to detect that the original expression evaluates to a boolean. Why hasn't it...?